From b1e403d69004a02dc6187c897e3ad5bd2a946edc Mon Sep 17 00:00:00 2001
From: Shivendra Pratap <quic_spratap@quicinc.com>
Date: Thu, 14 Mar 2024 21:18:05 +0530
Subject: [PATCH 3/4] QCLINUX: firmware: qcom_scm_addon: Remove implementation
 of qcom_scm_disable_sdi

Remove implementation of qcom_scm_disable_sdi as there are
no external users of this scm call.

Signed-off-by: Shivendra Pratap <quic_spratap@quicinc.com>
Upstream-Status: Pending
---
 drivers/firmware/qcom_scm_addon.c            | 23 +-------------------
 include/linux/firmware/qcom/qcom_scm_addon.h |  5 +----
 2 files changed, 2 insertions(+), 26 deletions(-)

diff --git a/drivers/firmware/qcom_scm_addon.c b/drivers/firmware/qcom_scm_addon.c
index f20680a70956..ca53ed8ce809 100644
--- a/drivers/firmware/qcom_scm_addon.c
+++ b/drivers/firmware/qcom_scm_addon.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0-only
 /*
- * Copyright (c) 2023 Qualcomm Innovation Center, Inc. All rights reserved.
+ * Copyright (c) 2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
  */
 
 #define QCOM_SCM_MP_CP_SMMU_APERTURE_ID         0x1b
@@ -547,27 +547,6 @@ int qcom_scm_sec_wdog_trigger(void)
 }
 EXPORT_SYMBOL_GPL(qcom_scm_sec_wdog_trigger);
 
-/**
- * qcom_scm_disable_sdi() - Disable SDI
- */
-void qcom_scm_disable_sdi(void)
-{
-	int ret;
-	struct qcom_scm_desc desc = {
-		.svc = QCOM_SCM_SVC_BOOT,
-		.cmd = QCOM_SCM_BOOT_WDOG_DEBUG_PART,
-		.owner = ARM_SMCCC_OWNER_SIP,
-		.args[0] = 1,
-		.args[1] = 0,
-		.arginfo = QCOM_SCM_ARGS(2),
-	};
-
-	ret = qcom_scm_call_atomic(__scm ? __scm->dev : NULL, &desc, NULL);
-	if (ret)
-		pr_err("Failed to disable secure wdog debug: %d\n", ret);
-}
-EXPORT_SYMBOL_GPL(qcom_scm_disable_sdi);
-
 /**
  * qcom_scm_spin_cpu(void) - spin on cpu
  */
diff --git a/include/linux/firmware/qcom/qcom_scm_addon.h b/include/linux/firmware/qcom/qcom_scm_addon.h
index f59aa2e1f6c3..5bfbb91e6cc5 100644
--- a/include/linux/firmware/qcom/qcom_scm_addon.h
+++ b/include/linux/firmware/qcom/qcom_scm_addon.h
@@ -1,6 +1,6 @@
 /* SPDX-License-Identifier: GPL-2.0-only */
 /*
- * Copyright (c) 2023 Qualcomm Innovation Center, Inc. All rights reserved.
+ * Copyright (c) 2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
  */
 
 #ifndef __QCOM_SCM_ADDON_H
@@ -43,7 +43,6 @@ extern int qcom_scm_invoke_callback_response(phys_addr_t out_buf,
 			unsigned int *data);
 extern int qcom_scm_sec_wdog_deactivate(void);
 extern int qcom_scm_sec_wdog_trigger(void);
-extern void qcom_scm_disable_sdi(void);
 extern int qcom_scm_enable_shm_bridge(void);
 extern int qcom_scm_delete_shm_bridge(u64 handle);
 extern int qcom_scm_create_shm_bridge(u64 pfn_and_ns_perm_flags,
@@ -174,8 +173,6 @@ static inline int qcom_scm_sec_wdog_trigger(void)
 	return -EPERM;
 }
 
-static inline void qcom_scm_disable_sdi(void) { }
-
 static inline int qcom_scm_enable_shm_bridge(void)
 {
 	return -EPERM;
-- 
2.34.1

